src/radical/pilot/agent/radical-pilot-agent-multicore.py

```
@@ -1620,15 +1619,20 @@ def lrms_config_hook(cls, name, cfg, lrms, logger):
 
     # --------------------------------------------------------------------------
     #
-    def construct_command(self, task_exec, task_args, task_numcores,
-                          launch_script_hop, opaque_slots):
+    def construct_command(self, cu, launch_script_hop):
+
+        opaque_slots = cu['opaque_slots']
+        cud          = cu['description']
+        task_exec    = cud['executable']
+        task_args    = cud.get('arguments')
+        task_cores   = cud.get('cores', 1)
```

I rather have it fail here if it is unset.